permissions/has_data_leak_rights.py

```diff
@@ -0,0 +1,44 @@
+import io
+import yaml
+
+from rest_framework import permissions
+
+with io.open('/omniport/apps/faculty_profile/config.yml') as stream:
+    CONFIGURATION = yaml.safe_load(stream)
+
+
+class CanDataLeak(permissions.BasePermission):
+    """
+    Custom permission to check if user can get faculty data
+    """
+
+    def has_permission(self, request, view):
+        """
+        Permission to get faculty data
+        """
+
+        # Check if dataLeak is configured or not
+        dataLeak = CONFIGURATION.get('dataLeak', None)
+
+        if dataLeak is not None:
+            app_token = dataLeak.get('facappToken', None)
+            valid_ips = dataLeak.get('ipAddresses', None)
+
+            if app_token is None or valid_ips is None:
+                return False
+
+            # request's ip address
+            x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+            if x_forwarded_for:
+                request_ip = x_forwarded_for
+            else:
+                request_ip = request.META.get('REMOTE_ADDR')
+
+            # request's authorization tokens
+            request_token = request.headers.get('Authorization')
+
+            if request_token == app_token:
+                if request_ip in valid_ips:
```

Comment:
If valid_ips is registered as let's say 5, `valid_ips` will be equal to `5` here. This will throw an Iterable error unless yaml loaders convert everything to string. Just check this once.

Reply:
But in the config.yml we are defining `ipAddresses` as an array of strings. Then this case of only `5` will be `['5']`